File: api/src/main/java/pl/alpheratzteam/obfuscator/util/RandomUtil.java
package pl.alpheratzteam.obfuscator.util;

import java.util.Random;

/**
 * @author Unix
 * @since 21.03.2020
 */

public final class RandomUtil
{
    private static final Random random = new Random();

    private RandomUtil() {
    }

    public static int nextInt(int max) throws IllegalArgumentException {
        return random.nextInt(max);
    }

    public static int nextInt() {
        return random.nextInt();
    }

    public static int nextInt(int min, int max) throws IllegalArgumentException {
        return random.nextInt(max - min + 1) + min;
    }

    public static boolean hasChance(double chance) {
        return Math.random() * 100.0 <= chance;
    }

    public static boolean nextBoolean() {
        return random.nextInt() == 0;
    }
}

File: src/main/kotlin/pl/alpheratzteam/obfuscator/util/RandomUtil.kt
package pl.alpheratzteam.obfuscator.util

import java.util.concurrent.ThreadLocalRandom

/**
 * @author Unix
 * @since 17.12.2020
 */

object RandomUtil {

    private val random = ThreadLocalRandom.current()

    /**
     * Randomizes a number.
     * @param max as max int range.
     * @return [Int] randomized int value.
     */
    fun int(max: Int) = random.nextInt(max)

    /**
     * Randomizes a number.
     * @return [Int] randomized int value.
     */
    fun int() = random.nextInt()

    /**
     * Randomizes a number.
     * @return [Double] randomized double value.
     */
    fun double() = random.nextDouble()

    /**
     * Randomizes a number.
     * @param min as min value to randomize.
     * @param max as max value to randomize.
     * @return [Int] randomized int value.
     */
    fun int(min: Int, max: Int) = random.nextInt(min, max)

    /**
     * Randomizes a number.
     * @param [min] as min value to randomize.
     * @param [max] as max value to randomize.
     * @return [Double] randomized double value.
     */
    fun double(min: Double, max: Double) = random.nextDouble(min, max)

    /**
     * Randomizes the chance.
     * @param chance as chance to check.
     * @return [Boolean], if `true` chance is right.
     */
    fun chance(chance: Double) = Math.random() * 100.0 <= chance

    /**
     * Randomizes a boolean.
     * @return [Boolean] randomized boolean value.
     */
    fun boolean() = random.nextBoolean()

}